src/attrib.d

```diff
@@ -611,23 +624,91 @@ public:
 extern (C++) final class AlignDeclaration : AttribDeclaration
 {
 public:
-    uint salign;
+    Expression ealign;
+    structalign_t salign = STRUCTALIGN_DEFAULT;
 
-    extern (D) this(uint sa, Dsymbols* decl)
+    extern (D) this(Loc loc, Expression ealign, Dsymbols* decl)
     {
         super(decl);
-        salign = sa;
+        this.loc = loc;
+        this.ealign = ealign;
     }
 
     override Dsymbol syntaxCopy(Dsymbol s)
     {
         assert(!s);
-        return new AlignDeclaration(salign, Dsymbol.arraySyntaxCopy(decl));
+        return new AlignDeclaration(loc,
+            ealign.syntaxCopy(), Dsymbol.arraySyntaxCopy(decl));
     }
 
     override Scope* newScope(Scope* sc)
     {
-        return createNewScope(sc, sc.stc, sc.linkage, sc.protection, sc.explicitProtection, this.salign, sc.inlining);
+        return createNewScope(sc, sc.stc, sc.linkage,
+            sc.protection, sc.explicitProtection, this,
+            sc.inlining);
+    }
+
+    override void setScope(Scope* sc)
+    {
+        //printf("AlignDeclaration::setScope() %p\n", this);
+        if (ealign && decl)
+            Dsymbol.setScope(sc); // for forward reference
+        return AttribDeclaration.setScope(sc);
```

Comment:
No. `AttribDeclaration.setScope` does not set `_scope` intentionally because it's not used for other attributes - ProtDeclaration, ConditionDeclaration + (VersionCondition, DebugCondition), etc.